interface/ceed-basis.c

```diff
@@ -540,6 +540,41 @@ int CeedQRFactorization(Ceed ceed, CeedScalar *mat, CeedScalar *tau,
   return 0;
 }
 
+/**
+  @brief Return Symmetric QR Factorization of matrix
+
+  @param[out] mat     Row-major matrix to be factorized in place
+  @param[out] lambda  Vector of length m of eigenvalues
+  @param n            Number of rows/columns
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Utility
+**/
+int CeedSymmetricQRFactorization(Ceed ceed, CeedScalar *mat, CeedScalar *lambda,
+                                 CeedInt n) {
+  return CeedError(ceed, 1, "Not implemented");
+}
+
+/**
+  @brief Return Simultaneous Diagonalization of two matrices
```

Comment:
Description should specify necessary predicates and possibly be described in terms of generalized eigenvalue problems.